File: haskel/glue/src/Glue/Lib/Math/Trigonometric.hs
module Glue.Lib.Math.Trigonometric where

import Glue.Env qualified as E
import Glue.Eval (Eval)
import Glue.IR (Frame, IR (..), Native (..))
import Glue.Lib.Math.Trigonometric.Acos qualified as Acos
import Glue.Lib.Math.Trigonometric.Asin qualified as Asin
import Glue.Lib.Math.Trigonometric.Atan qualified as Atan
import Glue.Lib.Math.Trigonometric.Cos qualified as Cos
import Glue.Lib.Math.Trigonometric.Sin qualified as Sin
import Glue.Lib.Math.Trigonometric.Tan qualified as Tan

trigonometric :: Frame Eval
trigonometric =
    E.frameFromList
        [ ("sin", Native (Func Sin.sin))
        , ("cos", Native (Func Cos.cos))
        , ("tan", Native (Func Tan.tan))
        , ("asin", Native (Func Asin.asin))
        , ("acos", Native (Func Acos.acos))
        , ("atan", Native (Func Atan.atan))
        ]

File: dart/glue/lib/src/lib/math/trigonometric.dart
import 'package:glue/src/module.dart';
import 'package:glue/src/lib/math/trigonometric/acos.dart';
import 'package:glue/src/lib/math/trigonometric/asin.dart';
import 'package:glue/src/lib/math/trigonometric/atan.dart';
import 'package:glue/src/lib/math/trigonometric/cos.dart';
import 'package:glue/src/lib/math/trigonometric/sin.dart';
import 'package:glue/src/lib/math/trigonometric/tan.dart';

/// Trigonometric module - trigonometric functions (radians)
/// Mirrors Haskell Glue.Lib.Math.Trigonometric exactly

/// The trigonometric module containing trigonometric functions
/// Mirrors Haskell Glue.Lib.Math.Trigonometric.trigonometric exactly
final ModuleInfo trigonometricModule = nativeModule('ffi.math.trigonometric', [
  // Direct trigonometric functions
  ('sin', sin),
  ('cos', cos),
  ('tan', tan),

  // Inverse trigonometric functions
  ('asin', asin),
  ('acos', acos),
  ('atan', atan),
]);
